### PR TITLE
VZ-3568.  Upgrade/upgrade-test fixes

### PR DESCRIPTION
**Pull request by michael-cico**
# Description

Upgrade/upgrade-test fixes.
- Revert impl for IsReleaseInstalled to previous impl that doesn't do fine-grained status checks
- Separate out new impl into new fn, IsReleaseDeployed for now, will debug and consolidate later if possible
- Set terminationGracePeriodSeconds to 0 on the platform operator deployment
- Handle error where setting UpgradeComplete status returns an error, and requeue for re-processing
- Fix up delays/waits in upgrade test pipeline
- Add unit test case for status update error on UpgradeComplete

Fixes VZ-3568.  

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by pfmackin — APPROVED** [review]

**Review by ddrechse — APPROVED** [review]

File: platform-operator/internal/helm/helmcli.go [hunk 1]
```diff
@@ -172,8 +172,8 @@ func IsReleaseFailed(releaseName string, namespace string) (bool, error) {
 	return releaseStatus == ChartStatusFailed, nil
 }
 
-// IsReleaseInstalled returns true if the release is installed
-func IsReleaseInstalled(releaseName string, namespace string) (found bool, err error) {
+// IsReleaseDeployed returns true if the release is installed
+func IsReleaseDeployed(releaseName string, namespace string) (found bool, err error) {
```
Comment (pfmackin) [inline platform-operator/internal/helm/helmcli.go:176]: comment should say if the release is deployed

File: platform-operator/controllers/verrazzano/upgrade.go [hunk 2]
```diff
@@ -51,9 +51,11 @@ func (r *Reconciler) reconcileUpgrade(log *zap.SugaredLogger, cr *installv1alpha
 	msg := fmt.Sprintf("Verrazzano upgraded to version %s successfully", cr.Spec.Version)
 	log.Info(msg)
 	cr.Status.Version = targetVersion
-	err := r.updateStatus(log, cr, msg, installv1alpha1.UpgradeComplete)
+	if err := r.updateStatus(log, cr, msg, installv1alpha1.UpgradeComplete); err != nil {
+		return ctrl.Result{Requeue: true}, err
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/upgrade.go:55]: should use newRequeueWithDelay()

File: platform-operator/controllers/verrazzano/component/helm/helm_component.go [hunk 3]
```diff
@@ -120,7 +120,7 @@ func (h HelmComponent) IsInstalled(_ *zap.SugaredLogger, _ clipkg.Client, namesp
 // IsReady Indicates whether or not a component is available and ready
 func (h HelmComponent) IsReady(log *zap.SugaredLogger, client clipkg.Client, namespace string) bool {
 	ns := resolveNamespace(h, namespace)
-	installed, _ := helm.IsReleaseInstalled(h.ReleaseName, resolveNamespace(h, namespace))
+	installed, _ := helm.IsReleaseDeployed(h.ReleaseName, resolveNamespace(h, namespace))
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/helm/helm_component.go:123]: variable should be deployed

**Review by pfmackin — DISMISSED** [review]
Few minor comments you can fix next go around

File: platform-operator/internal/helm/helmcli.go (hunk 1 above)
Comment (michael-cico) [reply platform-operator/internal/helm/helmcli.go:176]: ✅

File: platform-operator/controllers/verrazzano/upgrade.go (hunk 2 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/upgrade.go:55]: ✅

File: platform-operator/controllers/verrazzano/component/helm/helm_component.go (hunk 3 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/helm/helm_component.go:123]: ✅

**Review by pfmackin — APPROVED** [review]

**Pull request merged**